### PR TITLE
ci: report test coverage to coveralls.io

### PR DESCRIPTION
**Pull request by oxtoacart**
This generates a test coverage report and uploads it to coveralls.io

Updates #cleanup

### CODE REVIEW

**Review by clairew — APPROVED** [review]
looks awesome!

**Comment by oxtoacart** [comment]
I realized that courtney wasn't actually doing anything. Now that I'm close to having courtney actually running, I'm having an issue because all executions of the `go` command end up printing out a bunch of debug stuff like this:

```
+ repo_root=/home/runner/work/tailscale/tailscale/tool/../tool/gocross/../..
+ cd /home/runner/work/tailscale/tailscale/tool/../tool/gocross/../..
+ toolchain=/home/runner/.cache/tailscale-go
+ [[ -d /home/runner/.cache/tailscale-go ]]
++ grep -E '^go ' go.mod
++ cut -f2 -d.
+ want_go_minor=21
+ have_go_minor=
+ [[ -f /home/runner/.cache/tailscale-go/VERSION ]]
++ head -1 /home/runner/.cache/tailscale-go/VERSION
++ cut -f2 -d.
+ have_go_minor=21
+ have_go_minor=21
+ [[ -z 21 ]]
+ [[ 21 -lt 21 ]]
+ [[ ! -d /home/runner/.cache/tailscale-go ]]
+ gocross_path=gocross
+ gocross_ok=0
++ git rev-parse HEAD
+ wantver=[11](https://github.com/tailscale/tailscale/actions/runs/6939094186/job/18875868472?pr=10340#step:8:12)cd0cfe62c[12](https://github.com/tailscale/tailscale/actions/runs/6939094186/job/18875868472?pr=10340#step:8:13)5b5ebc08623585a759a9fb[19](https://github.com/tailscale/tailscale/actions/runs/6939094186/job/18875868472?pr=10340#step:8:20)[26](https://github.com/tailscale/tailscale/actions/runs/6939094186/job/18875868472?pr=10340#step:8:27)4
+ [[ -x gocross ]]
++ gocross gocross-version
```

courtney runs `go list` and it doesn't know what to do with all that extra output ...

File: tool/gocross/gocross-wrapper.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 set -euo pipefail
 
-if [[ "${CI:-}" == "true" ]]; then
+if [[ "${CI:-}" == "true" && "${NOBASHDEBUG:-}" != "true" ]]; then
```
Comment (oxtoacart) [inline tool/gocross/gocross-wrapper.sh:11]: I had to do this to make courtney happy, as it executes `go` and can't handle the bash debug output from `set -x`.

**Comment by oxtoacart** [comment]
Thanks for the review @clairew ! I feel pretty comfortable about the changes I made post-review, so I'm merging.

**Pull request merged**